templates/new/batch_runner.py.txt

```diff
@@ -1,21 +1,26 @@
 import logging
-from surround import Runner
-from stages import {safe_project_name}Data
+from surround import Runner, RunMode
+from stages import AssemblerState
 
 logging.basicConfig(level=logging.INFO)
 
 class BatchRunner(Runner):
-    def run(self, is_training=False):
-        self.assembler.init_assembler(True)
-        data = {safe_project_name}Data()
+
+    def load_data(self, mode, config):
+        # Create the state
+        state = AssemblerState()
 
         # Load data to be processed
         raw_data = "TODO: Load raw data here"
 
-        # Setup input data
-        data.input_data = raw_data
+        # Setup input data into the state
+        state.input_data = raw_data
+
+        return state
 
-        # Run assembler
-        self.assembler.run(data, is_training)
+    def run(self, mode=RunMode.BATCH_PREDICT):
```

Comment:
This function doesn't need to be in the generated code. We can document that this function exists and can be overridden if desired but it isn't needed for the simple cases.

Reply:
Done :)